File: js/Day1/sonar-sweep-part-2.js
const fs = require("fs");
const depths = fs.readFileSync("./js/Day1/input.txt").toString().split("\r\n");

let WINDOW_SIZE = 3;
let measurement = 0;
for (let i = 0; i < depths.length - WINDOW_SIZE; i++) {
  let prevDepth = depths
    .slice(i, i + WINDOW_SIZE)
    .reduce((sum, current) => sum + Number(current), 0);
  let currDepth = depths
    .slice(i + 1, i + WINDOW_SIZE + 1)
    .reduce((sum, current) => sum + Number(current), 0);
  if (currDepth > prevDepth) {
    measurement++;
  }
}

console.log({ measurement });

File: rust/Day1/sonar_sweep_part_2.rs
use std::env;
use std::fs;

struct Sonar;
impl Sonar {
    pub fn sweep(depths: Vec<i32>) -> i32 {
        let window_size = 3;
        let mut measurement = 0;
        for i in 0..depths.len() - window_size {
            let prev = depths[i..i + window_size].iter().sum::<i32>();
            let cur = depths[i + 1..i + window_size + 1].iter().sum::<i32>();
            if cur > prev {
                measurement += 1;
            }
        }

        measurement
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let filename = &args[1];
    println!("In file {}", filename);
    let contents = fs::read_to_string(filename)
        .expect("Something went wrong reading the file");
    
    let depths: Vec<i32> = contents.split("\r\n").filter_map(|w| w.parse().ok()).collect();
    let count = Sonar::sweep(depths);
    println!("{:?}", count);
}
